test/e2e/kappcontroller/migrate_managed_name_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package kappcontroller
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware-tanzu/carvel-kapp-controller/test/e2e"
+)
+
+func Test_Managed_Name_App_Migration(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+	oldName := "simple-app-ctrl"
+	name := "simple-app"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", oldName})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	c := `---
```

Comment:
nit: typically in golang single letter names are only used for method receivers. lets call it a bit longer e.g. `config`